src/JsonSchema/Constraints/Constraint.php

```diff
@@ -125,10 +124,14 @@ protected function incrementPath(JsonPointer $path = null, $i)
      * @param JsonPointer|null $path
      * @param mixed            $i
      */
-    protected function checkArray($value, $schema = null, JsonPointer $path = null, $i = null)
+    protected function checkArray(&$value, $schema = null, JsonPointer $path = null, $i = null, $coerce = false)
```

Comment:
Ah, good eye, thank you. I've added phpdocs, following the patterns and style already in place.

As for adding $coerce as a `Factory` flag, well, that was the approach I tried previously. It made sense in my previous approach because I was driving everything through `check`, and it just became a `mode` flag. The problem there was that I had no reference to work with in `check`, so I couldn't properly coerce in all situations. 

Now that I'm driving it through an explicit function call, `coerce`, it's no longer really a configuration value. If one were to, say, call `check` immediately after calling `coerce`, I would just have to set any stored fields back to false, so it would be another moving part with no real value.

I was actually interested in your reaction to what I did in `CollectionConstraint`. Any thoughts?